operator/cmd/mesh/uninstall.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/operator/pkg/cache"
+	"istio.io/istio/operator/pkg/helmreconciler"
+	"istio.io/istio/operator/pkg/util/clog"
+	"istio.io/istio/operator/pkg/util/progress"
+	proxyinfo "istio.io/istio/pkg/proxy"
+	"istio.io/pkg/log"
+)
+
+type uninstallArgs struct {
+	// kubeConfigPath is the path to kube config file.
```

Comment:
- Would it be possible to have a `--filename` flag that takes an IstioOperator?  The user might have two installed Istios?
- Would it be possible to have `--profile-from-cluster` as suggested by https://github.com/istio/istio/issues/23443 so that we can ask to uninstall?  For example, if the user installed the demo profile into 'istio-system-v16' we would want the default to be to uninstall that, rather than the defaults.

Reply:
added --filename and --set flag option to align with the install path. the --profile-from-cluster can be discussed later separately.